File: tools-excel/src/main/java/cn/gjing/tools/excel/valid/DefaultNumericValidation.scala
package cn.gjing.tools.excel.valid

import org.apache.poi.ss.usermodel.Sheet
import org.apache.poi.ss.util.CellRangeAddressList

/**
 * @author Gjing
 **/
class DefaultNumericValidation extends ExcelValidation{
  /**
   * Custom data validation rules
   *
   * @param numericValid NumericValid
   * @param sheet        sheet
   * @param firstRow     First row
   * @param firstCol     First col
   * @param lastCol      Last col
   */
  override def valid(numericValid: NumericValid, sheet: Sheet, firstRow: Int, firstCol: Int, lastCol: Int): Unit = {
    val helper = sheet.getDataValidationHelper
    val numericConstraint = helper.createNumericConstraint(numericValid.validationType.getType, numericValid.operatorType.getType, numericValid.expr1,
      if ("" == numericValid.expr2) null else numericValid.expr2)
    val regions = new CellRangeAddressList(firstRow, if (numericValid.boxLastRow == 0) firstRow
    else numericValid.boxLastRow + firstRow, firstCol, lastCol)
    val dataValidation = helper.createValidation(numericConstraint, regions)
    dataValidation.setShowErrorBox(numericValid.showErrorBox)
    dataValidation.setShowPromptBox(numericValid.showPromptBox)
    dataValidation.setErrorStyle(numericValid.rank.getRank)
    dataValidation.createErrorBox(numericValid.errorTitle, numericValid.errorContent)
    dataValidation.createErrorBox(numericValid.errorTitle, numericValid.errorContent)
    sheet.addValidationData(dataValidation)
  }
}

File: tools-excel/src/main/java/cn/gjing/tools/excel/valid/DefaultNumericValidation.java
package cn.gjing.tools.excel.valid;

import org.apache.poi.ss.usermodel.DataValidation;
import org.apache.poi.ss.usermodel.DataValidationConstraint;
import org.apache.poi.ss.usermodel.DataValidationHelper;
import org.apache.poi.ss.usermodel.Sheet;
import org.apache.poi.ss.util.CellRangeAddressList;

/**
 * Default value verifier
 *
 * @author Gjing
 **/
public class DefaultNumericValidation implements ExcelNumericValidation {
    @Override
    public void valid(ExcelNumericValid excelNumericValid, Sheet sheet, int firstRow, int lastRow, int firstCol, int lastCol) {
        DataValidationHelper helper = sheet.getDataValidationHelper();
        DataValidationConstraint numericConstraint = helper.createNumericConstraint(excelNumericValid.validationType().getType(),
                excelNumericValid.operatorType().getType(), excelNumericValid.expr1(), "".equals(excelNumericValid.expr2()) ? null : excelNumericValid.expr2());
        CellRangeAddressList regions = new CellRangeAddressList(firstRow, lastRow,firstCol, lastCol);
        DataValidation dataValidation = helper.createValidation(numericConstraint, regions);
        dataValidation.setShowErrorBox(excelNumericValid.showErrorBox());
        dataValidation.setErrorStyle(excelNumericValid.rank().getRank());
        dataValidation.createErrorBox(excelNumericValid.errorTitle(), excelNumericValid.errorContent());
        dataValidation.createErrorBox(excelNumericValid.errorTitle(), excelNumericValid.errorContent());
        sheet.addValidationData(dataValidation);
    }

}
